src/amber/router/parsers/optionals.cr

```diff
@@ -138,8 +136,8 @@ module Amber::Router::Parsers
       unless optional_end
         carat_position = 0
 
-        path[0..optional_start].each do |segment|
-          carat_position += segment.size
+        path[0..optional_start].each do |s|
+          carat_position += s.size
```

Comment:
This is the only change that I don't care for...what's the benefit to switching these to single letter names?

Reply:
It was due to an ameba failure that it was shadowing the var `segment` higher in the file.

Reply:
Hm. I think ameba and I disagree about that -- segment on line 121 is declared and used inside the loop. Oh well, I guess.

Reply:
I have a problem with that rule too, maybe we could disable it globally in `.ameba.yml`?

Reply:
Btw, this could be rewritten to `carat_position += path[0..optional_start.sum(&.size)`